CONTRIBUTING.md

```diff
@@ -138,6 +138,18 @@ Following these guidelines will help us to merge your pull requests smoothly:
    create a better end product. Committers will wait until after you've removed
    the WIP prefix to merge your changes.
 
+### Downstream Dependencies
+
+If the contents of your pull request has a dependency on a downstream project, i.e.
```

Comment:
I do not believe OMR changes should have dependencies on downstream projects as it should be the other way around. If a change is being contributed to OMR the contributor may ask the OMR committers to wait for their downstream project to be ready but the OMR committers should also make other downstream projects aware of the changes that are coming so they can also be ready.

Dependencies and the requirements on downstream projects as I see them:
1. A new feature/API is added to OMR and it is completely self contained.  No work required by downstream projects to consume and take advantage of this feature.
2. A new feature/API is added to OMR but requires downstream projects to configure or provide support for it to be useful. This *should* not break downstream projects but until they are ready to provide the required extra bits they can not take advantage of the new feature.
3. An API change/removal which will break all downstream projects using the component/feature containing the change.  This will require downstream projects to modify their code to move to the latest version of OMR.  This *will* break consumers until they have updated.  This is where co-ordination of the OMR project committers/leads is required to ensure that all downstream projects are aware of the upcoming changes and facilitate, where possible, to help the downstream consumers update.  These types of changes will require a lot more oversight as there must be a very good reason for this.  In a lot of cases the `old` API and the `new` API can live side by side for a release to ensure downstream projects have enough time to update.

Reply:
Fixed in https://github.com/eclipse/omr/pull/2306/commits/b003075004f5b75061a6073c3671f9a7730eda60